Sources/AWSLambdaEvents/APIGateway+V2.swift

```diff
@@ -117,3 +119,80 @@ extension APIGateway.V2 {
         }
     }
 }
+
+// MARK: - Codable Request body
+
+extension APIGateway.V2.Request {
+    /// Generic body decoder for JSON payloads
+    ///
+    /// Example:
+    /// ```
+    /// struct Request: Codable {
+    ///   let value: String
+    /// }
+    ///
+    /// func handle(context: Context, event: APIGateway.V2.Request, callback: @escaping (Result<APIGateway.V2.Response, Error>) -> Void) {
+    ///   do {
+    ///     let request: Request? = try event.decodedBody()
+    ///     // Do something with `request`
+    ///     callback(.success(APIGateway.V2.Response(statusCode: .ok, body:"")))
+    ///   }
+    ///   catch {
+    ///     callback(.failure(error))
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// - Throws: `DecodingError` if body contains a value that couldn't be decoded
+    /// - Returns: Decoded payload. Returns `nil` if body property is `nil`.
+    public func decodedBody<Payload: Codable>(decoder: JSONDecoder = JSONDecoder()) throws -> Payload? {
```

Comment:
rename Payload -> Body

Reply:
I wonder if instead of `decoder: JSONDecoder = JSONDecoder()` which should do something like `decoder: (Body) -> String` and have a default impl that uses JSONDecoder

Reply:
following on that line of thinking, @fabianfett maybe knows if this should be `(Body) ->[UInt8]`, or String is good enough

Reply:
The decoder should be `(String) -> Body`, since we are decoding the content of the `body` property, which is `String?`. Will update.

Reply:
I've given this a shot, but kind of defeats the purpose of the pull request.

Here is what the extension would look like:
```swift
public func decodedBody<Body: Codable>(decoder: (String.UTF8View) throws -> Body) throws -> Body? {
    guard let utf8 = body?.utf8 else {
        return nil
    }
    return try decoder(utf8)
}
```

And what the callee would look like:

Reply:
@eneko I meant the library would offer a default impl as well, for example (non optimized):

```swift
public func decodedBody<Body: Decodable>(decoder: @escaping (String) throws -> Body = Self.jsonDecoder) throws -> Body? {
  try self.body.map(decoder)
}
        
public static func jsonDecoder<T: Decodable>(json: String) throws -> T {
  try JSONDecoder().decode(T.self, from: Data(json.utf8))
}
```